lib/chef/provider/git.rb

```diff
@@ -218,21 +161,115 @@ def remote_matches?(remote_url, check_remote_command_result)
         check_remote_command_result.stdout.strip.eql?(remote_url)
       end
 
-      def current_revision_matches_target_revision?
-        (!@current_resource.revision.nil?) && (target_revision.strip.to_i(16) == @current_resource.revision.strip.to_i(16))
+      def target_revision
+        @target_revision ||= sha_hash?(@new_resource.revision) ? @new_resource.revision : remote_resolve_reference
       end
 
-      def target_revision
-        @target_revision ||= begin
-          if sha_hash?(@new_resource.revision)
-            @target_revision = @new_resource.revision
-          else
-            @target_revision = remote_resolve_reference
+      alias revision_slug target_revision
+
+      private
+
+      def run_options(run_opts = {})
+        run_opts[:group] = @new_resource.group if @new_resource.group
+        run_opts[:log_tag] = @new_resource.to_s
+        run_opts[:timeout] = @new_resource.timeout if @new_resource.timeout
+        run_opts[:user] = @new_resource.user if @new_resource.user
+        run_env = run_options_env
+        run_opts[:environment] = run_env unless run_env.empty?
+        run_opts
+      end
+
+      def run_options_env
+        env = {}
+        env_home = process_executor_home
+        env["HOME"] = env_home unless env_home.nil?
+        env["GIT_SSH"] = @new_resource.ssh_wrapper if @new_resource.ssh_wrapper
+        env.merge!(@new_resource.environment) if @new_resource.environment
+        env
+      end
+
+      def process_executor_home
+        env_home = nil
+        if @new_resource.user
+          # Certain versions of `git` misbehave if git configuration is
+          # inaccessible in $HOME. We need to ensure $HOME matches the
+          # user who is executing `git` not the user running Chef.
+          env_home = begin
+            require "etc"
+            Etc.getpwnam(@new_resource.user).dir
+          rescue ArgumentError # user not found
+            raise Chef::Exceptions::User, "Could not determine HOME for specified user '#{@new_resource.user}' for resource '#{@new_resource.name}'"
           end
         end
+        env_home
+      end
+
+      def cwd
+        @new_resource.destination
       end
 
-      alias :revision_slug :target_revision
+      def git(*args)
+        ["git", *args].compact.join(" ")
+      end
+
+      def git_standard_executor(args, run_opts = {})
+        git_command = git(args)
+        Chef::Log.debug "> #{git_command}"
```

Comment:
this should probably say "running #{git_command}" or something slightly more verbose